public/app/features/alerting/unified/components/rule-editor/ExpressionEditor.tsx

```diff
@@ -1,19 +1,57 @@
-import { TextArea } from '@grafana/ui';
-import React, { FC } from 'react';
+import React, { FC, useCallback, useMemo, useState } from 'react';
+import { CoreApp, DataQuery } from '@grafana/data';
+import { QueryEditorRenderer } from 'app/features/query/components/QueryEditorRenderer/QueryEditorRenderer';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { PromQuery } from 'app/plugins/datasource/prometheus/types';
+import { LokiQuery } from 'app/plugins/datasource/loki/types';
 
 interface Props {
   value?: string;
   onChange: (value: string) => void;
   dataSourceName: string; // will be a prometheus or loki datasource
 }
 
+const noop = () => {};
+
 // @TODO implement proper prom/loki query editor here
 export const ExpressionEditor: FC<Props> = ({ value, onChange, dataSourceName }) => {
+  const [query, setQuery] = useState<DataQuery>({ refId: 'A', hide: false });
```

Comment:
I think `value` should be added to query model as well?

Reply:
there is no value prop on DataQuery

Reply:
No, but we will map it from `value` to `expr`.